feat: Add variants and update stories for Popover

NatalieLoyzaga opened the pull request:
# Objective
<!-- Describe what this change achieves, and the details of how it works. -->

- Update Popover to include left and right Placement variant. 
- Allow for flip and fallback placements
- Update stories to show off clickable and hoverable story - Due to the makeup of this component being different to Tooltip, I decide to rather than re-invent the wheel but to illustration on the Kaizen Story of how you would use this component and make it clickable or hoverable.
- Include a 'Sticker sheet' story.

# Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

- After the work on Tooltips we decided Popover was lacking variants and functionality.
- We have updated both the component but the UI Kit

# Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/48232362/150023653-392e1e46-7d95-4346-a9f0-6a1970096d79.png)

![image](https://user-images.githubusercontent.com/48232362/150023718-0f3f1e41-cc1b-4b37-8952-031a6dea3b88.png)

![image](https://user-images.githubusercontent.com/48232362/150721829-d33c0789-a41d-4830-9400-da803d197021.png)

jmitsios commented:
@NatalieLoyzaga - Few small things I noticed:
_UI Kit design is on the right_
![image](https://user-images.githubusercontent.com/95595231/149706949-74c6a531-41c6-4bd7-bd74-627f78b738f0.png) ![image](https://user-images.githubusercontent.com/95595231/149706916-f500c051-3f66-4fad-ad6a-f296b7e418b0.png)
- Needs to tighten up the spacing between the header and the body text.
- What shadow size is being used?
- Space along the bottom is larger than the other sides, so we need to keep that consistent padding around all edges of the container.
- Could we see an example also with Icons next to the header please for the different moods?

jmitsios commented:
@NatalieLoyzaga 
- Looks to be missing the hover-state on the [x] close icon.
- _(Question for my knowledge)_ does the 'optional link' on Storybook have a way of inheriting our platforms link style + color?  Or does it just sit as the default browser hyperlink style?
- Sticker sheet page
  - is the only way to the popovers by clicking the 'Open all popovers' button? Could we also have them interactive/clickable individually?
  - Could we change the headers of each popover to match the mood (i.e: "Cautionary" instead of Header, etc..)

NatalieLoyzaga commented:
@jmitsios To respond to your last comment

- I have updated the button to have hover and focus state as per the UI kit.
- Sticker sheet...The way the component works which is different to Tooltip makes the sticker sheet story extremely convoluted. For every popover example it required its own component and it's own hook (controls). To be able to add the onClick functionality to each of them would then require additional controls to each of the individual hooks...I personally think it's a lot of extra code and rather the single button that controls all is a work around. ( of course if someone has a more autonomous idea/way of doing this I can explore )
- I have updated the headings.

lloydstubber commented on draft-packages/popover/KaizenDraft/Popover/styles.scss:
```diff
@@ -49,12 +45,13 @@ $large-width: 450px;
   border-radius: $border-solid-border-radius;
   color: $color-purple-800;
   text-align: left;
+  padding: $spacing-sm 1rem 1rem;
```
nit: I don't like how the UI Kit is using spacing that isn't in our variables 😬

NatalieLoyzaga replied:
The spacing is 12px in the UI kit but because of the line-height of the heading and body text, adding this 1rem padding around the whole container was the workaround to still getting a total spacing of 12px rather than adding additional padding in multiple spots. If that makes any sense.

lloydstubber commented on draft-packages/popover/KaizenDraft/Popover/styles.scss:
```diff
@@ -112,25 +109,32 @@ $negative-box-border-color: $color-red-300;
+  font-family: $typography-paragraph-body-font-family;
+  font-size: $typography-heading-6-font-size;
+  font-weight: $typography-heading-6-font-weight;
+  letter-spacing: $typography-heading-6-letter-spacing;
+  line-height: $typography-paragraph-small-line-height;
+  position: static;
+  margin-bottom: $spacing-xs;
   display: flex;
   align-items: center;
   white-space: nowrap;
+  padding-right: $spacing-md;
 }
 
 .container {
-  @include ca-inherit-baseline;
-  @include kz-typography-paragraph-small;
-  padding: ($ca-grid / 6) ($ca-grid / 2) ($ca-grid);
+  position: static;
+  font-family: $typography-paragraph-body-font-family;
+  font-size: $typography-heading-6-font-size;
+  font-weight: $typography-paragraph-small-font-weight;
+  letter-spacing: $typography-paragraph-small-letter-spacing;
+  line-height: $typography-paragraph-small-line-height;
```
Our type styles should be the one variable across the whole declaration. Guessing that first one is all `heading-6` and the second is all `paragraph-small`. If it looks incorrect in the browser we'll need to flag this in the UI Kit.

lloydstubber commented:
<img width="371" alt="Screen Shot 2022-01-21 at 10 58 09 am" src="https://user-images.githubusercontent.com/18339250/150441171-c2bc94d5-0e78-4595-9ef9-c3b343663275.png">

<img width="350" alt="Screen Shot 2022-01-21 at 10 58 40 am" src="https://user-images.githubusercontent.com/18339250/150441175-427da14c-6ad5-4944-9aac-89f6dcf68cf9.png">

Here's your focus state vs the UI Kit. Needs to be a bit tighter around the icon.

lloydstubber commented on draft-packages/popover/KaizenDraft/Popover/styles.scss:
```diff
@@ -170,6 +161,7 @@ $negative-box-border-color: $color-red-300;
     color: $color-purple-800;
     outline: $color-blue-500 $border-focus-ring-border-style
       $border-focus-ring-border-width;
+    outline-offset: calc($border-focus-ring-border-width * -1);
```
I can't seem to see this style in the browser... Is it working when you preview it from the Storybook link? @NatalieLoyzaga

NatalieLoyzaga replied:
@lloydstubber I think because buildkite is failing the storybook preview hasn't updated...grr.

lloydstubber replied:
Ahh true. Want to just throw a screenshot from your local dev here? Then I can give it a tick

NatalieLoyzaga replied:
Yep done. Sorry should of done that anyways.

lloydstubber commented:
LGTM! Just let me know when it's green and I'll tick it

lloydstubber reviewed (approved):
Nice one! Got there in the end
